content/blog/2021-08-24-pre-defined-models-exp.md

```diff
@@ -0,0 +1,352 @@
+to classify bees and ants. We'll use DVC to handle experiments for us and we'll
+compare the results of both models at the end.
+
+## Initialize the pre-trained model
+
+We'll be fine-tuning the AlexNet model and the SqueezeNet model to classify
+images of bees and ants. You can find the project we're working with in
+[this repo](https://github.com/iterative/pretrained-model-demo).
+
+In the `pretrained_model_tuner.py` file, you'll find the code that defines both
+the AlexNet and SqueezeNet models. We start by initializing these models so we
+can get the number of model features and the input size we need for fine-tuning.
+
+Since the project has everything we need to initialize the models, we can start
+training and comparing the differences between them with the ants/bees dataset.
+Running experiments to get the best tuning for each model can make it difficult
+to see which changes led to a better result. That's why will be using DVC to
+track changes in the code and the data.
+
+## Add a DVC pipeline
+
+Before we begin training, let's set up a DVC pipeline. We'll do that by running
+the following command.
+
+```dvc
+$ dvc init
+```
+
+This creates the `.dvc/` directory for configuration, the default cache
+location, and other necessary files and directories for DVC to work.
+
+Make sure you set up a
+[virtual environment](https://python.readthedocs.io/en/stable/library/venv.html)
+and install the dependencies with:
+
+```bash
+$ pip install -r requirements.txt
+```
+
+Now that you have DVC initialized, this is a good place to commit your changes.
+
+```git
+$ git status
+$ git add .
+$ git commit -m "added DVC pipeline"
+```
+
+With DVC in place, we need to add a stage to the pipeline so that we can train
+the model with DVC tracking each experiment.
+
+### Adding the train stage
+
+Stages in DVC let us define individual data processes and can be used to build
+detailed machine learning pipelines. You have the ability to define the
+different steps of model creation like preprocessing, featurization, and
+training.
+
+We'll add a stage for the training step in this example. To do that, run:
+
+```dvc
+$ dvc stage add --name train --deps data/hymenoptera_data --deps pretrained_model_tuner.py --metrics-no-cache --params lr,momentum results.json python pretrained_model_tuner.py
+```
+
+Let's break down the different parts of this command.
+
+- `--name` is the title of the stage
+- `--deps` defines the files and directories that a stage needs in order to run
+- `--params` defines the hyperparameters for the model
+- `--metrics-no-cache` specifies the metrics file produced by the stage, but DVC
+  doesn't track it
+
+After the metrics specification, there's the `python pretrained_model_tuner.py`
+and this defines the command for executing the training script.
+
+This will generate a new stage in the `dvc.yaml` file. If you take a look at it,
+you'll see something like:
+
+```yaml
```

Comment:
We can start with this YAML (and then briefly explain the helper command that we were using to generate it). Also it would be good to explain why do we need this file (for DVC to start managing data, understand where metrics to pay attention to, etc - some friendly explanations)

Reply:
I have the description of the `dvc stage add` command before this YAML. So you want me to move that description below the YAML? And I added friendlier explanations for each part of the command.

Reply:
yep, it can be simler to some extent - YAML is more or less friendly and something folks are already familiar vs some new command they haven't seen before. No strong opinion though. @dberenbaum wdyt? :)

Reply:
Yes, I think the first step could be cloning the repo and then explaining parts of the repo as necessary like `dvc.yaml`. I wouldn't repeat the steps that you've already done in that repo since it could be confusing and it takes longer to get to the transfer learning part. You can also give people the option to skip the explanations and jump right into the transfer learning part. I'm not even sure we really need a detailed explanation of the setup at the start of each blog post, but at least we should give the option of getting right to the main subject matter.